### PR TITLE
feat: upgrade to dotnet 8

### PR DESCRIPTION
**Pull request by kamilbaczek**

### CODE REVIEW

File: Chapter-1-initial-architecture/Src/Fitnet.UnitTests/Contracts/SignContract/SignContractTestData.cs [hunk 1]
```diff
@@ -2,30 +2,26 @@ namespace EvolutionaryArchitecture.Fitnet.UnitTests.Contracts.SignContract;
 
 internal sealed class SignContractTestData : IEnumerable<object[]>
```
Comment (meaboutsoftware) [inline Chapter-1-initial-architecture/Src/Fitnet.UnitTests/Contracts/SignContract/SignContractTestData.cs:3]: I adjusted it in chapter 2 to a bit different form - can you please adjust it here as well?

File: Chapter-1-initial-architecture/Src/Fitnet.IntegrationTests/Contracts/PrepareContract/PrepareContractTests.cs [hunk 2]
```diff
@@ -6,16 +6,14 @@ namespace EvolutionaryArchitecture.Fitnet.IntegrationTests.Contracts.PrepareCont
...
     IClassFixture<DatabaseContainer>
 {
-    private readonly HttpClient _applicationHttpClient;
-
-    public PrepareContractTests(WebApplicationFactory<Program> applicationInMemoryFactory,
-        DatabaseContainer database) =>
-        _applicationHttpClient = applicationInMemoryFactory
-            .WithContainerDatabaseConfigured(database.ConnectionString!)
-            .CreateClient();
+    private readonly HttpClient _applicationHttpClient = applicationInMemoryFactory
```
Comment (meaboutsoftware) [inline Chapter-1-initial-architecture/Src/Fitnet.IntegrationTests/Contracts/PrepareContract/PrepareContractTests.cs:14]: I did it a bit different in chapter 2 but will follow with your approach (will adjust it in separate branch)

**Review by meaboutsoftware — CHANGES_REQUESTED** [review]

File: Chapter-1-initial-architecture/Src/Fitnet.UnitTests/Contracts/SignContract/SignContractTestData.cs (hunk 1 above)
Comment (kamilbaczek) [reply Chapter-1-initial-architecture/Src/Fitnet.UnitTests/Contracts/SignContract/SignContractTestData.cs]: I just recall that for such case is cleaner why that we have currently implemented in chapter 2.
`internal sealed class SignContractTestData : TheoryData<DateTimeOffset, DateTimeOffset, DateTimeOffset, DateTimeOffset>
{
    public SignContractTestData()
    {
        AddRow(new DateTimeOffset(2021, 1, 1, 1, 1, 1, TimeSpan.Zero),
            new DateTimeOffset(2021, 1, 1, 1, 1, 1, TimeSpan.Zero),
            new DateTimeOffset(2021, 1, 1, 1, 1, 1, TimeSpan.Zero),
            new DateTimeOffset(2022, 1, 1, 1, 1, 1, TimeSpan.Zero));

        AddRow(new DateTimeOffset(2022, 6, 1, 12, 0, 0, TimeSpan.Zero),
            new DateTimeOffset(2022, 6, 1, 12, 0, 0, TimeSpan.Zero),
            new DateTimeOffset(2022, 6, 1, 12, 0, 0, TimeSpan.Zero),
            new DateTimeOffset(2023, 6, 1, 12, 0, 0, TimeSpan.Zero));
        
        AddRow(new DateTimeOffset(2023, 2, 15, 8, 30, 0, TimeSpan.Zero),
            new DateTimeOffset(2023, 2, 15, 8, 30, 0, TimeSpan.Zero),
            new DateTimeOffset(2023, 2, 15, 8, 30, 0, TimeSpan.Zero),
            new DateTimeOffset(2024, 2, 15, 8, 30, 0, TimeSpan.Zero));
    }
}`

Comment (kamilbaczek) [reply Chapter-1-initial-architecture/Src/Fitnet.UnitTests/Contracts/SignContract/SignContractTestData.cs]: @meaboutsoftware what you think. I could adjust in all chapter to this approach?

**Review by meaboutsoftware — APPROVED** [review]

**Pull request merged**